File: student-management-system/demo-student/src/main/java/com/example/demo/service/impl/StudentServiceImpl.java
package com.example.demo.service.impl;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.demo.mapper.StudentMapper;
import com.example.demo.model.Student;
import com.example.demo.service.StudentService;

@Service
public class StudentServiceImpl implements StudentService {
	
	private final StudentMapper studentMapper;
	
	@Autowired
	public StudentServiceImpl(StudentMapper studentMapper) {
		this.studentMapper = studentMapper;
	}

	@Override
	public List<Student> listStudent() {
		return studentMapper.selectList();
	}
	
	@Override
	public Student selectStudent(int id) {
		return studentMapper.selectStudent(id);
	}

	@Override
	public Integer addStudent(Student student) {
		return studentMapper.insertStudent(student);
	}

	@Override
	public Integer updateStudent(Student student) {
		return studentMapper.updateStudent(student);
	}

	@Override
	public Integer removeStudent(int id) {
		return studentMapper.deleteStudent(id);
	}

}

File: kotlin_learning/kotlin_librarian/src/main/kotlin/com/librarian/elibrarian/service/impl/StudentServiceImpl.kt
package com.librarian.elibrarian.service.impl

import com.librarian.elibrarian.model.Student
import com.librarian.elibrarian.repos.StudentRepository
import com.librarian.elibrarian.service.StudentService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class StudentServiceImpl(@Autowired private val studentRepository: StudentRepository): StudentService {
    override fun create(student: Student): Student {
        return studentRepository.saveAndFlush(student)
    }

    override fun list(): List<Student> {
        return studentRepository.findAll().toList()
    }

    override fun getById(studId: Int): Optional<Student> {
        return studentRepository.findById(studId)
    }

    override fun update(student: Student): Student {
        return studentRepository.saveAndFlush(student)
    }

    override fun remove(studId: Int): Int {
        var row: Int = 0
        row = try {
            studentRepository.deleteById(studId)
            1
        } catch(e: IllegalArgumentException) {
            e.printStackTrace()
            0
        }

        return row
    }
}